### PR TITLE
Stock Management - Month Details view

### PR DESCRIPTION
**Pull request by anvmn**
#642

### CODE REVIEW

**Comment by anvmn** [comment]
Peek 2023-03-26 11-43

**Comment by anvmn** [comment]
Selection_1721

**Comment by anvmn** [comment]
Selection_1725

**Comment by anvmn** [comment]
Selection_1724

**Review by mariano-dagostino — COMMENTED** [review]
@anvmn This PR seems a bit complex for me to review from a technical point of view. I'm adding @amitaibu for a more deep review.

Some comments however:

1. Are you considering the expired date as part of the stock calculation? If the shipment expires in 2 months, and the amount of stock provided let's say enough for 6 months do you show 2 months of stock (based on the expiration date?) or 6?
1. This [screenshot](https://github.com/TIP-Global-Health/eheza-app/pull/679#issuecomment-1484042804) shows a table without rows. Should we display some 'no results' table?

**Comment by anvmn** [comment]
@mariano-dagostino 

> 1. Are you considering the expired date as part of the stock calculation?

No, expiration date is for record only. No logic needs to be applied there.

> 2. Should we display some 'no results' table?

There were no such requirements, but it's not a complex development. @adamhstewart what do you think?

**Comment by anvmn** [comment]
> This PR seems a bit complex for me to review from a technical point of view. I'm adding @amitaibu for a more deep review.

@amitaibu would you like take a look?

File: client/src/elm/Backend/StockUpdate/Utils.elm [hunk 1]
```diff
@@ -1,9 +1,299 @@
...
+            Dict.values measurements.stockUpdate
+                |> List.sortWith (sortByDate .dateRecorded)
+                |> List.head
+                |> Maybe.map .dateRecorded
+                |> Maybe.withDefault
+                    -- If no stock update entries were found, we know that
+                    -- health center is yet to activate Stock Management
+                    -- feature.
+                    -- Indication of this would be setting first stock update
+                    -- momth to future date.
```
Comment (amitaibu) [inline client/src/elm/Backend/StockUpdate/Utils.elm]:
```suggestion
                    -- month to future date.
```

**Review by amitaibu — APPROVED** [review]
Yeah, this PR is a beast. So apart of a nitpick, MWG

**Pull request merged**